criu/proc_parse.c

```diff
@@ -1372,6 +1372,8 @@ static int parse_mountinfo_ent(char *str, struct mount_info *new, char **fsname)
 	cure_path(new->root);
 
 	root_link.len = strlen(new->root);
+	if (root_link.len >= PATH_MAX - 1)
+		goto err;
 	strcpy(root_link.name, new->root);
```

Comment:
We can print some meaningful error here about long root.

Also probably we should use strlcpy as it is "more overflow safe"? But I don't insist.

And please replace PATH_MAX with sizeof(root_link.name) which would be more robust.

Reply:
I will soon push a new version. Please have a look at the error message. If it makes any sense. Happy to change it to something else.

Reply:
Thanks! The message looks good.